### PR TITLE
Native quiver

### LINKED ISSUE #671: Native quiverplots
Quiver plots should be implemented as native quiver plots in `pgfplots`, i.e. similar to what is mentioned in its manual in Section 4.5.8 "Quiver Plots (Arrows)":
quiver
Section 4.6.10 "3D quiver plots (arrows)" states that 3D is supported by supplying a `w` option as well.

Our current output (`ACID(22)`) is:
```latex
\begin{tikzpicture}[%
arrow1/.style={->,color=mycolor1,solid}
]

\begin{axis}[%
width=0.951\figureWidth,
height=\figureHeight,
at={(0\figureWidth,0\figureHeight)},
scale only axis,
xmin=   0,
xmax=   1,
ymin=   0,
ymax=0.45,
axis background/.style={fill=white}
]
\addplot [arrow1] coordinates{(   0,   0) (0.45,0.45)};
\addplot [arrow1] coordinates{(   1,   0) (0.55,0.45)};
\end{axis}
\end{tikzpicture}%
```

### PR DESCRIPTION
**Pull request by egeerardyn**
This is a continuation of the work of @jam4375 in #679 to fix #671.

Some more personal TODOs before I consider this finished:
 - [x] use `opts_*` framework to set most of these options
 - [x] move arrow options from figure level to the plot level (this patch makes this more convenient)
 - [x] perform graphical tests in all environments (hashes should pass already)

### CODE REVIEW

**Comment by egeerardyn** [comment]
The output looks okay, not great for `ACID(22)`, but that one seems like an unnatural test case anyway. Some TODOs have been added to the test and code to signal this for other devs.

**Comment by egeerardyn** [comment]
@PeterPablo : Could you review this PR, please?

**Comment by PeterPablo** [comment]
Will do now.

**Comment by PeterPablo** [comment]
Okay...
* Hashes match on R2014a/b and octave
* Graphical output compilation struggles on my Linux machine. It might be caused by the old Lua version. Thus I temporarily changed to `pdflatex` in the `Makefile`:
```
There is no LUA colorspace for 'rgb' - skipping LUA color map definition of hot
There is no LUA colorspace for 'rgb' - skipping LUA color map definition of mymap
make[1]: *** [test10-converted.pdf] Error 1
```

R2014b:
image
image
* [ ] Shouldn't this be better now?

R2014a:
*will edit*

**Comment by egeerardyn** [comment]
I noticed `ACID(22)` as well, but since the example is quite contrived, I didn't inspect it any further. But I think that `ACID(24)` is unacceptable, I seem to have missed that one. So we should fix the scaling before merging this PR.

File: src/matlab2tikz.m [hunk 1]
```diff
@@ -3676,35 +3673,56 @@ case size(cData,2) % specific y-data is given
...
+        arrowHeadOpts = opts_new();
+        arrowHeadOpts = opts_add(arrowHeadOpts, 'scale length', ...
+                                 '{max(0.01,\pgfplotspointmetatransformed/1000)}');
+        arrowHeadOpts = opts_add(arrowHeadOpts, 'scale width', ...
+                                 '{0.5*max(0.01,\pgfplotspointmetatransformed/1000)}');
+        headStyle = ['-{Straight Barb[' opts_print(m2t, arrowHeadOpts, ',') ']}'];
+        quiverOpts = opts_add(quiverOpts, 'every arrow/.append style', ...
+                              ['{' headStyle '}']);
+    else
+        %TODO: check what we can do for scaling non-barbed arrows
```
Comment (PeterPablo) [inline src/matlab2tikz.m]: Is there even anything we *need* to do? If `ShowArrowHead` is `false`, there is no arrow head to scale/style?!

Comment (egeerardyn) [inline src/matlab2tikz.m]: Indeed, there is nothing we can do. I must have misinterpreted the width option to mean the line width, but it's really the width of the arrow. Will remove that TODO.

File: src/matlab2tikz.m [hunk 2]
```diff
@@ -3676,35 +3673,56 @@ case size(cData,2) % specific y-data is given
...
-        sprintf('arrow%d/.style', m2t.quiverId), ...
-        ['{', arrowOptions, '}']);
-    % - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-    % return the vector field code
-    str = [str, ...
-        sprintf(['\\',name,' [arrow',num2str(m2t.quiverId), '] ', ...
-        'coordinates{(',format,') (',format,')};\n'],...
-        data)];
-    %FIXME: external
+    if showArrowHead
```
Comment (PeterPablo) [inline src/matlab2tikz.m]: This is, where the octave codepath diverges. (octave output of `testGraphical([22])` looks good)

**Comment by PeterPablo** [comment]
The octave graphical output is fine as well and code looks good to me. So why not merge this now, so that the other PRs can already be merged with the updated hashes/preamble and fix the remaining MATLAB issues (remove TODO note, fix MATLAB scaling) separately?

File: src/matlab2tikz.m [hunk 3]
```diff
@@ -3676,35 +3673,54 @@ case size(cData,2) % specific y-data is given
...
+        %TODO: scale the arrows more rigorously to match MATLAB behavior
+        %There is a "MaxHeadSize" property (at least in R2014b) that plays a
+        %role in determining the quiver size.
+        arrowHeadOpts = opts_new();
+        arrowHeadOpts = opts_add(arrowHeadOpts, 'scale length', ...
```
Comment (PeterPablo) [inline src/matlab2tikz.m]: This leads to wrong output of `ACID(22)`. Manually setting `scale width` and `scale length` to 10 results in better output. Consequently there must be a bug in the two formulas. Rereading the pgfmanual concerning `\pgfplotspointmetatransformed` might be a good start (didn't do this myself).

**Pull request merged**